debian/patches/10_endless_make-soname-be-the-major-version

```diff
@@ -0,0 +1,32 @@
+Description: Endlessm.com patch: Make soname be the major version
+ ndn ships with soname == version - that is going to make a lot of packages,
+ change that so that soname matches only the major version.
+ .
+ We'll bump the version manually
+ .
+ libndn-cxx (0.5.0+git20161122-2) unstable; urgency=medium
+ .
+   * New Upstream release
+Author: Niv Sardi <xaiki@debian.org>
+
+---
+The information above should follow the Patch Tagging Guidelines, please
+checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
+are templates for supplementary fields that you might want to add:
+
+Origin: Endless
+Bug: https://phabricator.endlessm.com/T13932
+Forwarded: https://github.com/named-data/ndn-cxx/pull/7
+Last-Update: 2016-12-10
+
+--- libndn-cxx-0.5.0+git20161122.orig/wscript
++++ libndn-cxx-0.5.0+git20161122/wscript
+@@ -184,7 +184,7 @@ def build(bld):
+     if bld.env.enable_shared:
+         bld(features=['pch', 'cxx', 'cxxshlib'],
+             vnum=VERSION_BASE,
+-            cnum=VERSION_BASE,
++            cnum=VERSION_BASE.split('.')[0],
```

Comment:
i really implemented this by forcing a dep on the same source version in subsequent packages.
having minor version doesn't help us much (because if they break ABI on micro, which they have done, it doesn't solve any issue, and it doesn't serve the purpose to be able to integrate back from debian once they are stable)